File: org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/extension/ui/NotificationBadgeStatus.java
/*
 * Copyright (c) 2010-2023 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
package org.eclipse.scout.rt.client.extension.ui;

import org.eclipse.scout.rt.platform.status.IStatus;
import org.eclipse.scout.rt.platform.status.Status;

public class NotificationBadgeStatus extends Status {
  private static final long serialVersionUID = 1L;

  public NotificationBadgeStatus(String notificationBadgeText) {
    super(notificationBadgeText, IStatus.INFO);
  }

  @Override
  public int compareTo(IStatus o) {
    if (o instanceof NotificationBadgeStatus) {
      return super.compareTo(o);
    }
    return -1;
  }
}

File: eclipse-scout-core/src/status/NotificationBadgeStatus.ts
/*
 * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
import {InitModelOf, Status, strings} from '../index';
import $ from 'jquery';

export class NotificationBadgeStatus extends Status {

  constructor(model?: InitModelOf<Status>) {
    super($.extend(true, {severity: Status.Severity.INFO}, model));
  }

  override cssClass(): string {
    return strings.join(' ', Status.cssClassForSeverity(this.severity), 'notification-badge');
  }

  /**
   * @returns {NotificationBadgeStatus} a clone of this Status instance.
   */
  override clone(): Status {
    let modelClone = $.extend({}, this);
    return new NotificationBadgeStatus(modelClone);
  }

  override equals(o: any): boolean {
    if (!(o instanceof NotificationBadgeStatus)) {
      return false;
    }
    return super.equals(o);
  }
}
